### ISSUE TITLE
There is a high probability that mail configuration SMTP will fail to send mail, and the prompt is very vague

### ISSUE THREAD

**Issue by letwang** [PrestaShop/PrestaShop#29098]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

Now, due to the use of the Swift mailing class library, the sendmail provided in the background is basically equivalent to a dummy, which has no effect on the virtual host user, because the service provider will not allow command line permissions to be enabled.
For the last SMTP method, when the configuration is incorrect, the prompt is not clear enough, so the user has no basis for investigation.
Therefore, I suggest that the specific email error report is thrown to tell the front-end user, which is convenient for troubleshooting and improving the success rate of email configuration.

https://github.com/PrestaShop/PrestaShop/issues/29076#issuecomment-1186502106

### Steps to reproduce

![image](https://user-images.githubusercontent.com/3644168/179443202-0c062474-3b9d-48d7-8bfe-22bfaa5d5b59.png)

### PrestaShop version(s) where the bug happened

1.7.X

### PHP version(s) where the bug happened

7.4.X

**Comment by letwang** [comment]
The world-renowned WordPress can still use the simple and friendly PHP built-in function mail to send emails, so I can't understand why SwiftMail manufacturers have the right to choose to use PHP's built-in function mail because of the so-called security barbaric castration.
Unfortunately the great PrestaShop uses SwiftMail, so PrestaShop users also lose the option to use PHP's built-in function mail.
There are countless posts on the PrestaShop forum due to the failure of users to configure emails, which has greatly damaged the reputation of PrestaShop. As a world-friendly and convenient e-commerce software, it should not make it so complicated for users to get started.
Security should be considered by technology, and should not be passed on to customers. Please return the customer's right to choose to use PHP's built-in function mail.

**Comment by matks** [comment]
> Security should be considered by technology, and should not be passed on to customers.

This is exactly what we are doing. PrestaShop is a technology, not a product. To be more accurate it is an open source project. Security is our number 1 priority as an open source project for many reasons, and one is that the loss due to stolen data or hacks is a lot more damaging than inconvenience.

But I have good news: you can use the technology and customize it for your needs this is the strenght of CMS. So if you disagree with SwiftMailer opinion there is a very simple open source friendly way for you to achieve this: simply customize PrestaShop mail sending behavior to use the solution you think suitable.

If you do not have the skills to do it, you can hire a developer to build it for you as a module 👍 . You could even sell this module to other PrestaShop users that, just like you, disagree with SwiftMailer vision.

So basically:
- PrestaShop is the technology platform built for all users
- If you disagree with one of the choices implemented inside PrestaShop, you can change this - the code is open and it is a CMS, so you have plenty customization capabilities
- Customers (= people willing to invest) can pay for modules, themes and customizations and build the very product they want using the platform as a foundation

I hope this will help you find the way to use prestashop as you wish 😉

**Comment by letwang** [comment]
Then I only have the last choice, the option of sendmail, but this option is unavailable for many people, because the virtual host company will not allow us to open permission to execute the command line.

**Comment by hibatallahAouadni** [comment]
Hello @letwang 

We only use GitHub issues to discuss about bugs and new features in the PrestaShop project. If you need support, you can find resources to help you on this page: [Get help with PrestaShop](https://www.prestashop-project.org/support/).

Alternatively, you can also ask for help in the [community forums](https://www.prestashop.com/forums/) or in the [public Slack channel](https://prestashop.slack.com).

Thank you

**Comment by matks** [comment]
> Then I only have the last choice, the option of sendmail, but this option is unavailable for many people, because the virtual host company will not allow us to open permission to execute the command line.

No, you can still use `mail()` but you have to implement the transport yourself

See discussion https://github.com/swiftmailer/swiftmailer/issues/866

I think one of the last comments is a good answer:

> If you really need such a transport, you are still free to implement it on your own (you could even just take the existing code, the license allows that). Then it is your responsibility to deal with security issues, but not that of the maintainers of a widely used library who decided not to bear this after considering the consequences.

> The MailTransporter was deprecated because of security reasons. We don't want to keep something that will always have security issues.

@hibatallahAouadni I think the initial post was a feature request and then we also started a discussion 😉

**Comment by hibatallahAouadni** [comment]
Hello @matks 

Thanks for your quick replies :pray: 
> @hibatallahAouadni I think the initial post was a feature request and then we also started a discussion 😉 

Well, @letwang choose the Bug template and she doesn't respect the provided template, so that's why I close the issue as invalid.

Thanks!

**Comment by letwang** [comment]
oh my gosh~
It would be a shame to turn a blind eye to PrestaShop's questions because my questioning pose was not elegant enough~
I actually express 3 questions above:
1. The built-in function MAIL in PHP cannot be used, so many users cannot use PrestaShop products normally, which affects the reputation of PrestaShop. PrestaShop forum so many mail question posts is a good proof, and the world-renowned WordPress has not abandoned the PHP built-in function MAIL. (It is recommended to add support for MAIL)
2. The provided sendmail method is equivalent to no effect for the virtual host user, because the virtual host company will not open the command line execution permission for the user, and it is unbelievable to expect the user to develop the MAIL by himself. You want to change all the users. Are you a repairman? (It is recommended to add support for MAIL)
3. It is easy to report errors after SMTP configuration. I found the solution in the above picture. It only needs to return the error report to the front end, which is convenient for users to check the SMTP error report, and finally improves the success rate of SMTP configuration mail. Unfortunately, the official website turns a blind eye to the exception thrown error, so that the user does not know why the SMTP error is reported, and eventually the email cannot be sent. (It is recommended to optimize the error message and return the error to the user to let the user know the error)

![image](https://user-images.githubusercontent.com/3644168/179641230-81dce14b-a222-421c-9e04-a7d1d8732139.png)

**Comment by hibatallahAouadni** [comment]
Hello @letwang 

> It would be a shame to turn a blind eye to PrestaShop's questions because my questioning pose was not elegant enough~
I actually express 3 questions above:

You can ask your questions here in the [public Slack channel](https://prestashop.slack.com).

Thank you
